### PR TITLE
fixes #836: Simplify regex expressions

### LINKED ISSUE #836: Simplify regexes
It has come to my attention that executing a regex of the following form: ".*.*.*.*.*.*.*bla" or ".*?.*?.*?.*?.*?.*?.*?bla" can have serious performance issues.  I found that there is an exponential growth relative to the number of "/*" or "/*?" in the regex and to the size of the value in terms of performance.  We have discovered this regex  running against 1000+ character values to actually take days!  This has caused significant issue in terms of hanging up tservers.  This is obviously a user error, however the system should be able to account for this.  We need a mechanism to simplify this regex to avoid the performance implications.

### PR DESCRIPTION
**Pull request by ivakegg**
* Created a NodeTransform to simplify regex expressions for performance reasons

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java [hunk 1]
```diff
@@ -112,7 +117,24 @@ public void regexPushdownAnyfieldTransformRuleTest() throws Exception {
...
+
+    @Test
+    public void regexSimplifierTransformRuleTest() throws Exception {
+        // @formatter:off
+        String query = "BLA == '.*?.*?x' && " +
+                "BLA =~ 'ab.*.*' && " +
+                "BLA =~ 'a.*.*.*.*?.*?' && " +
+                "BLA =~ '.*?.*?.*bla.*?.*?blabla' && " +
+                "_ANYFIELD_ =~ '.*.*?.*?<bla>'";
+        String expected = "BLA == '.*?.*?x' && " +
```
Comment (FineAndDandy) [inline warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java:129]: shouldn't this be simplified to `BLA == '.*?x'`?

Comment (ivakegg) [reply warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java:129]: That is an equals node, not a regex node.

Comment (ivakegg) [reply warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java:129]: That is an equals node, not a regex node.

Comment (FineAndDandy) [reply warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java:129]: ha, right. You got me

**Pull request merged**